assembly/src/ast/parsers/constants.rs

```diff
@@ -0,0 +1,262 @@
+                    stack.push(Operation::get_operator(&current_char));
+                }
+            }
+        } else {
+            // push character of the number (or its name) to the `parsed_number` string
+            parsed_value.push(current_char);
+        }
+
+        if next_char.is_none() {
+            break;
+        }
+    }
+
+    // push the remaining number to the postfix expression
+    if !parsed_value.is_empty() {
+        let parsed_number = parse_number(op, expression, constants, parsed_value)?;
+        postfix_expression.push(parsed_number);
+    }
+
+    // push remaining on the stack operators to the postfix expression
+    while let Some(element) = stack.pop() {
+        postfix_expression.push(element);
+    }
+
+    Ok(postfix_expression)
+}
+
+/// Evaluates constant expression represented in postfix notation
+fn evaluate_postfix_expression(
+    op: &Token,
+    expression: &str,
+    postfix_expression: Vec<Operation>,
+) -> Result<Felt, ParsingError> {
+    let mut stack = Vec::new();
+
+    for operation in postfix_expression.iter() {
+        match operation {
+            // if the operation is a value
+            Operation::Value(value) => stack.push(*value),
+            // if the operation is an operator
+            _ => {
+                let right = stack.pop().expect("stack is empty");
+                let left = stack.pop().expect("stack is empty");
+                stack.push(compute_statement(left, right, operation, op, expression)?);
+            }
+        }
+    }
+
+    // get the result from the stack
+    stack.pop().ok_or_else(|| {
+        ParsingError::invalid_const_value(
+            op,
+            expression,
+            &format!("constant expression {} is incorrect", op),
+        )
+    })
+}
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+/// Returns the number in `value` or the constant value if the value is the name of the constant.
+fn parse_number(
+    op: &Token,
+    expression: &str,
+    constants: &LocalConstMap,
+    value: String,
+) -> Result<Operation, ParsingError> {
+    let parsed_number = value.parse::<u64>();
+    // if the parsed value is a number push it on the stack
+    if let Ok(parsed_number) = parsed_number {
+        Ok(Operation::Value(Felt::new(parsed_number)))
+    }
+    // if it is a name of the constant get its value from the `constants` map
+    else {
+        let parsed_number = constants.get(&value).ok_or_else(|| {
+            ParsingError::invalid_const_value(
+                op,
+                expression,
+                &format!("constant with name {} was not initialized", value),
+            )
+        })?;
+        Ok(Operation::Value(Felt::new(*parsed_number)))
+    }
+}
+
+/// Returns `true` if th left operator has higher priority than the right, `false` otherwise.
+fn left_has_greater_priority(left: &char, right: &Operation) -> bool {
```

Comment:
minor nit: the term usually used "precedence" rather than "priority" (i.e, "operator precedence").